p2p/proto/block.proto

```diff
@@ -0,0 +1,66 @@
+syntax = "proto3";
+import "p2p/proto/common.proto";
+import "google/protobuf/timestamp.proto";
+
+// for now, we assume a small consensus, so this fits in 1M. Else, these will be repeated
+message Signatures {
+    BlockID id = 1;
+    repeated Signature signatures = 2; // can be more explicit here about the signature structure as this is not part of account abstraction
+}
+
+// Note: commitments may change to be for the previous blocks like comet/tendermint
+// hash of block header sent to L1
+message BlockHeader {
+    BlockID parent_block = 1;
+
+    google.protobuf.Timestamp time = 2; // TODO: see if this needs to be Felt252 or can be converted
+
+    Address sequencer_address = 3;
+
+    Merkle state_diffs = 4; //  By order of (contract, key), taking last in case of duplicates.
+                            // This means the proposer needs to sort after finishing the block (TBD: patricia? )
+
+    // State is optional and appears every X blocks for the last block. This is to support
+    // snapshot sync and also so that light nodes can sync on state without state diffs.
+    Merkle state = 5; // hash of contract and class patricia tries. Same as in L1. Later more trees will be included
+
+    Hash proof_fact = 6; // for Kth block behind. A hash of the output of the proof
+
+    // The following merkles can be built on the fly while sequencing/validating txs.
+    Merkle transactions = 7; // By order of execution
+
+    Merkle events = 8; // By order of issuance.
+
+    Merkle receipts = 9; // By order of issuance.
+
+    uint32 protocol_version = 10;
+
+    ChainID chain_id = 11;
+}
+
+message BlockProof {
+    bytes proof = 1; // proof size is currently 142K
+}
+
+// sent to all peers (except the ones this was received from, if any).
+// for a fraction of peers, also send the GetBlocks response (as if they asked for it for this block)
+message NewBlock {
+    BlockID id = 1;
+}
+
+// result is (BlockHeader, StateDiff*)*
```

Comment:
What about block bodies?

Reply:
A block body is just a series of statediff messages, a proof message, and signatures message.

Reply:
Does a node not need the tx?

Reply:
A node that is not a validator, no. Our scale comes from the fact that "follower" nodes will not need to execute transactions. They trust the consensus, after a few blocks there's a proof, and after some time, an L1 state update. 

A node may want the transactions for statistics, or to be able to reexecute a block for debugging or to show a trace.

Reply:
What about the validator? How do they get the tx?

Reply:
A node that is a validator gets the txs, a node that is not trusts the consensus until it gets a proof. The concept of SN is that since there are proofs we need smaller consensus.
Note that a node can decide to listen to the consensus messages and get the txs and validate if it wishes.

Reply:
Yea, I mean, the spec need to have the ability to download txs as the validator node need it. It could be another endpoint.

Reply:
This spec doesn't include the consesus messages.

Reply:
Still there should be way to download a block's transactions outside of consensus context. A full node will need to serve transaction data, how will it obtain that for existing blocks?